File: CaprezzoDigitale.WebApi/Models/Messaggio.cs
namespace CaprezzoDigitale.WebApi.Models
{
    public class Messaggio
    {
        public long Id { get; set; }
        public string Titolo { get; set; }
        public string Sottotitolo { get; set; }
        public DateTime DataPubblicazione { get; set; }
        /// <summary>
        /// Questa property DEVE CONTENERE solo il nome del file (eventualmente sotto-directory).
        /// </summary>
        public string UrlImmagineCopertina { get; set; }
        /// <summary>
        /// Al click dell'immagine di copertina viene aperto il pdf relativo.
        /// Questo per una migliore esperienza d'uso.
        /// </summary>
        public string UrlPdfImmagineCopertina { get; set; }
        public string Testo { get; set; }
        public short TipoMessaggioId { get; set; }
        public TipoMessaggio TipoMessaggio { get; set; }
        public List<Allegato> Allegati { get; set; }
    }
}

File: public-client/src/app/routes/dashboard/models/messaggio.ts
import { Allegato } from "./allegato";
import { TipoMessaggio } from "./tipo-messaggio";

export class Messaggio {
    id!: number;
    titolo!: string;
    sottotitolo!: string;
    dataPubblicazione!: Date;
    urlImmagineCopertina!: string;
    urlPdfImmagineCopertina!: string;
    testo!: string;
    tipologiaPostId!: number;
    tipoMessaggio!: TipoMessaggio;
    allegati!: Allegato[];

    constructor(data?: Partial<Messaggio>) {
        Object.assign(this, data);
    }
}
